Local grapl consul dns

twunderlich-grapl opened the pull request:
<!-- Thank you for your contribution to Grapl! Please take some time
to fill out this short template to help us review and understand your
PR. -->

### Which issue does this PR correspond to?
https://github.com/grapl-security/issue-tracker/issues/887
<!-- Please provide a link to the GitHub Issue this pull request is
meant to address. -->

### What changes does this PR make to Grapl? Why?
This sets up consul dns on port 53 inside nomad containers. ie from within a container you can now dig redis.service.consul and get an answer.
<!-- Please describe the functional impact of these changes and the
rationale behind them. This will help us understand your approach to
solving the problem. -->

### How were these changes tested?
Exec into a running nomad container
apt-get update && apt-get Install dnsutils
dig redis.service.consul. ANY
You'll get a reply
<!-- Please describe how these changes were tested. There should be
sufficient detail that reviewers can replicate your testing
procedure. If the procedure is a manual one that's OK, your
description will help us work with you to get it automated. -->

Commit:
Set consul dns recursors to Google DNS

We're about to forward all nomad dns requests to consul, we need to
set a dns recursor so we know what to forward dns requests to when the
request isn't to a .consul domain

Commit:
Run dnsmasq as nomad task to forward dns to consul

We're using dnsmasq as a container to forward dns to consul. We're using
a container for local grapl, since I expect that there's some dns
customization on people's laptops.

Commit:
Set dns search shortcut to make querying easier

With this we shouldn't need to type out the full $SERVICE.service.consul
anymore, just hit $SERVICE. THis still needs to be tested

wimax-grapl commented on nomad/local/grapl-local-infra.nomad:
```diff
@@ -428,6 +428,57 @@ job "grapl-local-infra" {
+      driver = "docker"
+
+      config {
+        image      = "andyshinn/dnsmasq"
```
https://hub.docker.com/r/andyshinn/dnsmasq/ says we should use a different one

twunderlich-grapl replied:
I've fixed this, thanks for pointing this out

Commit:
Switch to recommended dockerhub host

The old dockerhub host moved to a new one. This one is actively
maintained. I've also switched to a tag version for better debugability

Commit:
Switch consul tracing to using a dns endpoint

Instead of relying on an IP address that may not hold true, we're using
a consul dns address.

Commit:
Add dns fallback in case Google DNS is SOL

While extremely rare, Google DNS can fail completely at times. Since
we can include a max of 3 dns providers, may as well include a fallback
at a separate provider, just in case.

Commit:
Revert "Set dns search shortcut to make querying easier"

This reverts commit d4a2ff6247f16203e7e4a1caf29da99299d2d21b.
We may revive this at a later point, but for now we'll enforce adding the .service.consul suffix

Commit:
Format files

christophermaier commented on nomad/local/nomad-agent-conf.nomad:
```diff
@@ -15,6 +15,11 @@ plugin "docker" {
       enabled = true
     }
+
+    # We need net_admin for dnsmasq to work. Everything else should be default
```
Out of curiosity, do you have a link to documentation or code that says what the default capabilities are?

Commit:
Document default cap allow_list

Clarify what is default and what is not, and make it easy to check that
list for updates

Commit:
Add commentary around dnsmasq options

Commit:
Fix flake8 failure
